adm_config_set.php

```diff
@@ -102,177 +102,32 @@
 	$t_type = $f_type;
 }
 
-switch( $t_type ) {
-	case CONFIG_TYPE_STRING:
-		$t_value = $f_value;
-		break;
-	case CONFIG_TYPE_INT:
-		$t_value = (integer)constant_replace( trim( $f_value ) );
-		break;
-	case CONFIG_TYPE_FLOAT:
-		$t_value = (float)constant_replace( trim( $f_value ) );
-		break;
-	case CONFIG_TYPE_COMPLEX:
-	default:
-		$t_value = process_complex_value( $f_value );
-		break;
+# Parse the value
+if( $t_type == CONFIG_TYPE_STRING ) {
+	# Return strings as is
+	$t_value = $f_value;
+} else {
+	try {
+		$t_parser = new ConfigParser( $f_value );
+		$t_value = $t_parser->parse( ConfigParser::EXTRA_TOKENS_IGNORE );
```

Comment:
As mentioned elsewhere, I reused this parser which I initially wrote for other purposes; the original code was more generic than this, and was designed to throw an exception.

As I was working through adapting it, I realized that in the specific usage of the admin config page, even though it is invalid for users to submit more than one value, it is more user-friendly to silently drop the extra stuff rather than choke on it. 

I decided to leave this as an option, in case we want to make use of it later (e.g. have an ajax to validate the value prior to actually submitting it). I'm fine with changing the default if you prefer.